addons/interaction/functions/fnc_getWeaponAttachmentsActions.sqf

```diff
@@ -49,8 +49,30 @@ params ["_unit"];
     // "detach" actions
     {
         if (_x isEqualTo "") then {continue};
+        private _originalAttachment = _x;
 
-        private _config = _cfgWeapons >> _x;
+        private _switchableAttachments = [_originalAttachment] call CBA_fnc_switchableAttachments;
+        private _convertToActions = [];
+        {
+            if (_x == _originalAttachment) then {continue};
```

Comment:
```suggestion
            if (_x == _originalAttachment) then {continue};
            if (getText(_CBA_PIPItems >> _x) == _originalAttachment) then {continue};
```
CBA PIP optics should be ignored too. CUP has many scopes like this (e.g. ACOG and SUSAT).